File: rb/features.rb
# AdressApiFrance SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module AdressApiFranceFeatures
  def self.make_feature(name)
    case name
    when "base"
      AdressApiFranceBaseFeature.new
    when "test"
      AdressApiFranceTestFeature.new
    else
      AdressApiFranceBaseFeature.new
    end
  end
end

File: py/features.py
# AdressApiFrance SDK feature factory

from feature.base_feature import AdressApiFranceBaseFeature
from feature.test_feature import AdressApiFranceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AdressApiFranceBaseFeature(),
        "test": lambda: AdressApiFranceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
